src/lib/zkapp.ts

```diff
@@ -643,38 +661,69 @@ async function callUnproved<S extends typeof SmartContract>(
     methodName as any,
     methodArguments
   );
+  selfParty.sign(zkappKey);
+  selfParty.body.incrementNonce = Bool(true);
   let tx = Mina.createUnsignedTransaction(() => {
     Mina.setCurrentTransaction({ parties: [selfParty], nextPartyIndex: 1 });
   });
-  let txJson = tx.toJSON();
-  if (zkappKey !== undefined)
-    txJson = Ledger.signOtherParty(txJson, zkappKey, 0);
+  let txJson = tx.sign().toJSON();
   return txJson;
 }
 
-async function signJsonTransaction(
+async function addFeePayer(
```

Comment:
We should purely functionalize here too -- this one is especially dangerous to use mutation because of the async/await

Reply:
done